lib/cynic.dart

```diff
@@ -17,31 +16,50 @@ const Service _googleDns = const Service.forIp('8.8.8.8');
 /// servers as proof of internet connectivity. You will need to implement custom
 /// functionality for an intranet.
 Future<bool> isOnline({Duration timeout}) {
-  return _googleDns.isOnline(timeout: timeout);
+  return Reachable.googleDns.isOnline(timeout: timeout);
 }
 
 /// Returns a future that completes with whether the internet is not reachable.
 ///
 /// See [isOnline] for full documentation.
 Future<bool> isOffline({Duration timeout}) async {
-  return _googleDns.isOffline(timeout: timeout);
+  return Reachable.googleDns.isOffline(timeout: timeout);
 }
 
 /// Platform-agnostic indication of a remote service being online or offline.
-abstract class Service {
+abstract class Reachable {
+
+  static const Reachable googleDns = const Reachable.ip('8.8.8.8');
+
   /// Returns a future that completes if all [services] are online.
-  static Future<bool> allOnline(
-    Iterable<Service> services, {
+  static Future<bool> anyOnline(Iterable<Reachable> services, {
     Duration timeout,
-  }) async =>
-      !(await Future.any(services.map((s) => s.isOffline(timeout: timeout))));
+  }) async {
+    final List<bool> executions = await Future.wait(
```

Comment:
I'd just drop the LHS type definitions:

1. It's wrong, it's actually `Iterable<bool>`.
2. Just let type inference pick it up.

We also don't want to wait to check all services, i.e. if the first service says `true` in 0.1s, lets complete:

```dart
final completer = new Completer<bool>();
void _maybeComplete(bool isOnline) {
  if (isOnline && !completer.isCompleted) {
    completer.complete(true);
  }
}
final futures = services.map((s) => s.isOnline(timeout: timeout));
for (final future in futures) {
  future.then(_maybeComplete);
}
Future.wait(futures).then((all) {
  _maybeComplete(false);
});
return completer.future;
```